src/test/Business/MetaQuestionController.test.js

```diff
@@ -0,0 +1,147 @@
+        await sequelize.sync({force: true}); // cleans the 'Users' table
+    })
+
+    test('add appendix', async () => {
+        try {
+            await mqController.getAppendix(appendixData.tag);
+            expect(false).toBeTruthy();
+        } catch (e) {
+            expect(e instanceof EMSError).toBeTruthy();
+            expect(e.errorCode).toBe(MQ_PROCESS_ERROR_CODES.APPENDIX_TAG_DOESNT_EXIST)
+        }
+
+        await mqController.createAppendix({
+            ...appendixData,
+            callingUser: lecturerCallingDetails
+        })
+        try {
+            const appendix = await mqController.getAppendix(appendixData.tag);
+            expect(appendix.getTag()).toBe(appendixData.tag);
+            expect(appendix.getTitle()).toBe(appendixData.title);
+            expect(appendix.getContent()).toBe(appendixData.content);
+            expect(appendix.getKeywords().sort())
+                .toStrictEqual(appendixData.keywords.sort());
+        } catch (e) {
+            expect(false).toBeTruthy();
+        }
+    })
+
+    test('edit appendix', async () => {
+        await mqController.createAppendix({
+            ...appendixData,
+            callingUser: lecturerCallingDetails
+        })
+
+        // edit keywords only
+        const editData = {
+            callingUser: lecturerCallingDetails,
+            tag: appendixData.tag,
+            keywords: [],
+        }
+        await mqController.editAppendix(editData);
+        let updatedAppendix = await mqController.getAppendix(editData.tag);
+        expect(updatedAppendix.getTag()).toBe(appendixData.tag);
+        expect(updatedAppendix.getTitle()).toBe(appendixData.title);
+        expect(updatedAppendix.getContent()).toBe(appendixData.content);
+        expect(updatedAppendix.getKeywords()).toStrictEqual(editData.keywords);
+
+        // edit title
+        editData.title = 'new title';
+        await mqController.editAppendix(editData);
+        updatedAppendix = await mqController.getAppendix(editData.tag);
+        expect(updatedAppendix.getTag()).toBe(appendixData.tag);
+        expect(updatedAppendix.getTitle()).toBe(editData.title);
+        expect(updatedAppendix.getContent()).toBe(appendixData.content);
+        expect(updatedAppendix.getKeywords()).toStrictEqual(editData.keywords);
+
+        // edit content
+        editData.content = 'new content';
+        await mqController.editAppendix(editData);
+        updatedAppendix = await mqController.getAppendix(editData.tag);
+        expect(updatedAppendix.getTag()).toBe(appendixData.tag);
+        expect(updatedAppendix.getTitle()).toBe(editData.title);
+        expect(updatedAppendix.getContent()).toBe(editData.content);
+        expect(updatedAppendix.getKeywords()).toStrictEqual(editData.keywords);
+    })
+
+    afterAll(async () => {
+        await sequelize.close();
+    });
+});
+/*
+describe('MetaQuestionController FAILURE tests', () => {
```

Comment:
Why to comment it?
It is good to have "bad path" tests

Reply:
I didn't add any tests there...

Reply:
I meant why not to do a test there...

Reply:
I was trying to finish work fast because it took too long anyway so I tested only things I wanted to make sure work properly.
Couldn't come up with any failure test to test